### PR TITLE
Oracles

### PR DESCRIPTION
**Pull request by polgreen**
this fails CI because there's a function that has an unreachable branch that doesn't return anything

### CODE REVIEW

File: src/theory/quantifiers/oracle_engine.h [hunk 1]
```diff
@@ -77,9 +79,18 @@ class OracleEngine : public QuantifiersModule
...
+
+ /** functions for minimal parsing. These will be removed */
+  Node get_hex_numeral(std::string in);
+  Node get_dec_numeral(std::string in);
+  Node get_bin_numeral(std::string in);
+  Node responseParser(std::string &in);
  private:
   /** The oracle functions (user-context dependent) */
   context::CDList<Node> d_oracleFuns;
+  bool consistencyCheckPassed=false;
```
Comment (ajreynol) [inline src/theory/quantifiers/oracle_engine.h:92]: Use `d_` prefix for data members.

File: src/theory/quantifiers/oracle_engine.h [hunk 2]
```diff
@@ -57,13 +57,15 @@ class OracleEngine : public QuantifiersModule
...
-  /** Declare oracle fun */
 
-  /** Call an oracle with a set of arguments **/
-  std::string callOracle(const std::string &binary_name, 
-                         const std::vector<std::string> &argv);
+  /** get binary from oracle interface */
+  std::string getBinaryName(const Node n);
 
-  void declareOracleFun(Node f);
+  /** Call an oracle with a set of arguments **/
```
Comment (ajreynol) [inline src/theory/quantifiers/oracle_engine.h:64]: Can we move this to a separate file?  Like `oracle_call.h / cpp` or something?

We could have a utility like OracleCaller  which caches the I/O from a single oracle.

The point here is that I'd like oracle_engine to not involve details on how the oracle is called, instead it should be just the algorithm.

File: src/theory/quantifiers/oracle_engine.h [hunk 3]
```diff
@@ -77,9 +79,18 @@ class OracleEngine : public QuantifiersModule
                                 Node& assume,
                                 Node& constraint,
                                 std::string& binName);
+
+
+ /** functions for minimal parsing. These will be removed */
```
Comment (ajreynol) [inline src/theory/quantifiers/oracle_engine.h:84]: Probably move these to the oracle_call file as suggested above.

File: src/theory/quantifiers/oracle_engine.cpp [hunk 4]
```diff
@@ -116,23 +207,72 @@ void OracleEngine::check(Theory::Effort e, QEffort quant_e) {
       continue;
     }
     currInterfaces.push_back(q);
-    Trace("oracle-engine-state") << "Interface: " << q << std::endl;
+    Trace("oracle-engine-state") << "Interface: " << q << " with binary name " << getBinaryName(q) << std::endl;
   }
-
+  bool all_fapps_consistent=true;
```
Comment (ajreynol) [inline src/theory/quantifiers/oracle_engine.cpp:212]: Nit: we use camelCase so this should be `allFappsConsistent`.

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: src/theory/quantifiers/oracle_engine.h (hunk 2 above)
Comment (polgreen) [reply src/theory/quantifiers/oracle_engine.h:64]: Sure. Though I imagine that, when/if we have a modularised parser that we can pull in, this utility will be a lot more concise

Comment (polgreen) [reply src/theory/quantifiers/oracle_engine.h]: moved, put it in src/util

**Review by ajreynol — APPROVED** [review]

**Pull request merged**